lib/rules/checked-requires-onchange-or-readonly.js

```diff
@@ -0,0 +1,124 @@
+ */
+
+'use strict';
+
+const ASTUtils = require('jsx-ast-utils');
+const isCreateElement = require('../util/isCreateElement');
+const report = require('../util/report');
+const docsUrl = require('../util/docsUrl');
+
+const messages = {
+  missingProperty: 'Checked should be used with either onChange or readOnly.',
+  exclusiveCheckedAttribute:
+    'Use either checked or defaultChecked but not both.',
+};
+
+const targetProps = ['checked', 'onChange', 'readOnly', 'defaultChecked'];
+const targetPropSet = new Set(targetProps);
+
+const defaultOptions = {
+  ignoreMissingProperties: true,
+  ignoreExclusiveCheckedAttribute: true,
+};
+
+function extractTargetProps(properties, keyName) {
+  return properties.reduce((acc, prop) => {
+    if (
+      prop[keyName]
+      && prop[keyName].name
+      && targetPropSet.has(prop[keyName].name)
+    ) {
+      acc.add(prop[keyName].name);
+    }
+    return acc;
+  }, new Set());
+}
+
+module.exports = {
+  meta: {
+    docs: {
+      description:
+        'Enforce using `onChange` or `readonly` attribute when `checked` is used',
+      category: 'Best Practices',
+      recommended: false,
+      url: docsUrl('checked-requires-onchange-or-readonly'),
+    },
+    messages,
+    schema: [{
+      additionalProperties: false,
+      properties: {
+        ignoreMissingProperties: {
+          type: 'boolean',
+        },
+        ignoreExclusiveCheckedAttribute: {
+          type: 'boolean',
+        },
+      },
+    }],
+  },
+  create(context) {
+    const options = Object.assign({}, defaultOptions, context.options[0]);
+
+    function reportMissingProperty(node) {
+      report(context, messages.missingProperty, 'missingProperty', {
+        node,
+      });
+    }
+
+    function reportExclusiveCheckedAttribute(node) {
+      report(
+        context,
+        messages.exclusiveCheckedAttribute,
+        'exclusiveCheckedAttribute',
+        {
+          node,
+        }
+      );
+    }
+
+    const checkAttributesAndReport = (node, propSet) => {
+      if (options.ignoreExclusiveCheckedAttribute && propSet.has('checked') && propSet.has('defaultChecked')) {
+        reportExclusiveCheckedAttribute(node);
+      }
+
+      if (
+        options.ignoreMissingProperties
+        && propSet.has('checked')
+        && !(propSet.has('onChange') || propSet.has('readOnly'))
+      ) {
+        reportMissingProperty(node);
+      }
+    };
+
+    return {
+      JSXOpeningElement(node) {
+        if (ASTUtils.elementType(node) !== 'input') {
+          return;
+        }
+
+        const propSet = extractTargetProps(node.attributes, 'name');
+        checkAttributesAndReport(node, propSet);
+      },
+      CallExpression(node) {
+        if (!isCreateElement(node, context)) return;
+
+        const [firstArg, secondArg] = node.arguments;
+        if (
+          !firstArg
+          || firstArg.type !== 'Literal'
+          || firstArg.value !== 'input'
+        ) {
+          return;
```

Comment:
seems like this line isn't covered?

Reply:
I guess so. Thank you for letting me know. I'll cover it soon